libraries/lib-numeric-formats/formatters/ParsedNumericConverterFormatter.cpp

```diff
@@ -0,0 +1,590 @@
+         return std::nullopt;
+
+      for (i = 0; i < mFields.size(); i++)
+      {
+         const auto pos = mFields[i].pos;
+         const auto digits = mFields[i].digits;
+
+         if (pos >= valueString.size() || pos + digits > valueString.size())
+            return std::nullopt;
+
+         long val;
+
+         const auto fieldStringValue =
+            valueString.Mid(mFields[i].pos, mFields[i].digits);
+
+         if (!fieldStringValue.ToLong(&val))
+            return std::nullopt;
+
+         if (mFields[i].frac)
+            t += (val / (double)mFields[i].base);
+         else
+            t += (val * (double)mFields[i].base);
+      }
+
+      t /= mScalingFactor;
+
+      if (mNtscDrop)
+      {
+         int t_int = (int)(t + .000000001);
+         double t_frac = (t - t_int);
+         int tenMins = t_int / 600;
+         double frames = tenMins * 17982;
+         t_int -= tenMins * 600;
+         int mins = t_int / 60;
+         int addMins = 0;
+         if (mins > 0)
+         {
+            frames += 1800;
+            addMins = mins - 1;
+         }
+         frames += addMins * 1798;
+         t_int -= mins * 60;
+         if (mins == 0) // first min of a block of 10, don't drop frames 0 and 1
+            frames += t_int * 30 + t_frac * 30.;
+         else
+         { // drop frames 0 and 1 of first seconds of these minutes
+            if (t_int > 0)
+               frames += 28 + (t_int - 1) * 30 + t_frac * 30.;
+            else
+               frames += t_frac * 30. - 2.;
+         }
+         t = frames * 1.001 / 30.;
+      }
+
+      return t;
+   }
+
+   double SingleStep(double value, int digitIndex, bool upwards) const override
+   {
+      const auto dir = upwards ? 1 : -1;
+      for (size_t i = 0; i < mFields.size(); i++)
+      {
+         if (
+            (mDigits[digitIndex].pos >= mFields[i].pos) &&
+            (mDigits[digitIndex].pos < mFields[i].pos + mFields[i].digits))
+         { // it's this field
+            if (value < 0)
+               value = 0;
+
+            value *= mScalingFactor;
+
+            const double mult = pow(
+               10., mFields[i].digits -
+                       (mDigits[digitIndex].pos - mFields[i].pos) - 1);
+
+            if (mFields[i].frac)
+            {
+               value += ((mult / (double)mFields[i].base) * dir);
+            }
+            else
+            {
+               value += ((mult * (double)mFields[i].base) * dir);
+            }
+
+            if (mNtscDrop)
+            {
+               if ((value - (int)value) * 30 < 2)
+               {
+                  if ((((int)value) % 60 == 0) && (((int)value) % 600 != 0))
+                  {
+                     value = (int)value + (dir > 0 ? 2. : -1.) / 30.;
+                  }
+               }
+            }
+
+            if (value < 0.)
+            {
+               value = 0.;
+            }
+
+            value /= mScalingFactor;
```

Comment:
Not obvious, but -- this introduces the change to `Adjust` that you mentioned in meeting.  Clamping the stepped value to  and scaling were done in the wrong sequence.

Please mention this subtletly in a commit comment.

You must have observed the bad interaction of scaling and min and max while trying to make things work for beats.  Was this also a bug in the spectral selection toolbar?

In fact, maybe it used to be, but see commits 24c2c6e, 0dd3aef where the setting of minimum and maximum were disabled.

Reply:
This came solely from the code observation

Reply:
You found it deductively?  Well that is good too, even better.